src/parser/postgresparser.cpp

```diff
@@ -607,6 +626,114 @@ expression::AbstractExpression* PostgresParser::AExprTransform(A_Expr* root) {
   return result;
 }
 
+// This function takes in a Postgres NullTest primnode and transfers
+// it into Peloton AbstractExpression.
+expression::AbstractExpression* PostgresParser::NullTestTransform(NullTest* root) {
+  if (root == nullptr) {
+    return nullptr;
+  }
+
+  UNUSED_ATTRIBUTE expression::AbstractExpression* result = nullptr;
+  UNUSED_ATTRIBUTE ExpressionType target_type = ExpressionType::OPERATOR_IS_NULL;
+  NullTestType nulltesttype = root->nulltesttype;
+  int type_id = static_cast<int>(target_type);
+
+
+  expression::AbstractExpression* arg_expr = nullptr;
+  expression::AbstractExpression* right_expr = nullptr;
+  switch (root->arg->type) {
+    case T_ColumnRef: {
+      arg_expr =
+          ColumnRefTransform(reinterpret_cast<ColumnRef*>(root->arg));
+      break;
+    }
+    case T_A_Const: {
+
+      arg_expr = ConstTransform(reinterpret_cast<A_Const*>(root->arg));
+      break;
+    }
+    case T_A_Expr: {
+
+      arg_expr = AExprTransform(reinterpret_cast<A_Expr*>(root->arg));
+      break;
+    }
+    case T_ParamRef: {
+
+      arg_expr = ParamRefTransform(reinterpret_cast<ParamRef *>(root->arg));
+      break;
+    }
+    case T_List: {  // 656
+
+      arg_expr = ConstTransform(reinterpret_cast<A_Const*>(root->arg));
+
+      break;
+    }
+    default: {
+      LOG_DEBUG("Arg expr of type %d not supported yet...\n",
```

Comment:
"throw NotImplementedException(StringUtil::Format("Arg expr of type %d not supported yet...\n", root->arg->type));" may be better here.